### DISCUSSION TITLE
Sharepoint rest api from custom handlebar

### DISCUSSION THREAD

**Discussion by ayblanchet** [microsoft-search/pnp-modern-search#1572] [Q&A] [answered]
Hi,

I would like to call the sharepoint rest api from custom handlebar created with pnp-modern-search-extensibility, but I don't know how get the webpart context, spHttpClient or any service from this custom handlebar.
Can you help me?
Thank you,

Best regards,

**Comment by FranckyC** [comment, marked as answer]
Hi @ayblanchet, you can't use `SPHtpClient` in a Handlebars helper. Use a web component instead with the following approach where `this._serviceScope` is the page root service scope allowing to resolve generic services like `SPHttpClient` and others:

```
 public async connectedCallback() {
 	...
        const spHttpClient = this._serviceScope.consume<SPHttpClient>(SPHttpClient.serviceKey);
	...
```

**Reply by ayblanchet** [reply]
Hi @FranckyC, Ok I see https://microsoft-search.github.io/pnp-modern-search/extensibility/custom_web_component/.
Thank for your help,
